### PR TITLE
Shipping section validation

### PR DESCRIPTION
**Pull request by AvnerGuyAvrahamCom**

### CODE REVIEW

File: src/Components/CheckoutModal/Sections/Delivery/AccordionItemComponent.tsx [hunk 1]
```diff
@@ -63,31 +55,38 @@ export const AccordionItemComponent: FC<IAccordionItemComponent> = ({
...
+                }}
+                value={item.name}
+              >
+                <Box w="full">
+                  <Flex justify="space-between" w="full">
+                    <Box>{item.title}</Box>
+                    <Box>
+                      {item.currency}
+                      {item.cost}
+                    </Box>
```
Comment (artemGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Delivery/AccordionItemComponent.tsx:81]: Maybe use <Flex /> or as="span" or <Text />

File: src/Components/CheckoutModal/Sections/Shipping/index.tsx [hunk 2]
```diff
@@ -27,89 +28,125 @@ const FormLabelStyles: FormLabelProps = {
...
+            onPlaceSelected={(place) => console.log(place)}
+          />
+          <FormInput
+            name="streetAddress"
+            label="Street Address"
+            type="text"
+            w={isMobile ? '59%' : '100%'}
+          />
+
+          <Box mx="0.5" />
```
Comment (artemGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Shipping/index.tsx:75]: You have at least 5 times `<Box mx="0.5" />` maybe create some variable or component for this?

File: src/Components/CheckoutModal/Sections/Shipping/index.tsx [hunk 3]
```diff
@@ -27,89 +28,125 @@ const FormLabelStyles: FormLabelProps = {
...
+            formlabelstyle={FormLabelStyles}
+            variant={EVariantCheckBox.Square}
+            name="marketing"
+          >
+            <Text fontWeight="400">
+              {massages.marketing}
+              <Text as="label" fontWeight="bold">
+                Terms
+              </Text>
+              <Text as="label"> and </Text>
```
Comment (artemGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Shipping/index.tsx:134]: add text to i18n (`messages`)

File: src/Components/CheckoutModal/index.tsx [hunk 4]
```diff
@@ -18,25 +18,34 @@ const CheckoutModal = () => {
...
-  const [isMobileMediaQuery] = useMediaQuery('(min-width: 48rem)');
+  const [isMobileMediaQuery] = useMediaQuery('(min-width: 50rem)');
 
   useEffect(() => {
     setIsMobile(isMobileMediaQuery);
   }, [setIsMobile, isMobileMediaQuery]);
 
-  const handleClick = () => {
+  const handleClick = (variant?: string) => {
+    console.log(variant);
```
Comment (artemGuyAvrahamCom) [inline src/Components/CheckoutModal/index.tsx:28]: remove `console.log`

File: src/Components/CheckoutModal/Sections/Shipping/index.tsx (hunk 2 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/index.tsx:75]: done

File: src/Components/CheckoutModal/index.tsx (hunk 4 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/index.tsx:28]: done

File: src/Components/CheckoutModal/Sections/Shipping/index.tsx (hunk 3 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/index.tsx]: done

File: src/Components/CheckoutModal/OrderSummary/index.tsx [hunk 5]
```diff
@@ -45,16 +45,16 @@ const OrderSummary: FC<BoxProps> = (props) => {
           <OrderItem key={i} item={order} currency={orderData.currency} />
         ))}
         <Divider {...dividerStyle} />
-        {specialOffers.map((offer) => (
-          <Box key={offer}>
+        {specialOffers.map((offer, i) => (
+          <Box key={i}>
```
Comment (yehorGuyAvrahamCom) [inline src/Components/CheckoutModal/OrderSummary/index.tsx:49]: Why did you change key to index? 
It's a bad practice to use index like key

File: src/Components/CheckoutModal/OrderSummary/index.tsx [hunk 6]
```diff
@@ -45,16 +45,16 @@ const OrderSummary: FC<BoxProps> = (props) => {
...
             <SpecialOffer offerName={offer} />
             <Divider {...dividerStyle} />
           </Box>
         ))}
         <Box mb="2" />
-        {generalDetails.map((detail) => (
+        {generalDetails.map((detail, i) => (
           <GeneralDetail
-            key={orderData[detail]}
+            key={i}
```
Comment (yehorGuyAvrahamCom) [inline src/Components/CheckoutModal/OrderSummary/index.tsx:57]: The same: Don't use index like key

File: src/Components/CheckoutModal/ProgressBar/ProgressControls.tsx [hunk 7]
```diff
@@ -48,8 +46,16 @@ export const ProgressControls: FC<IStepperControls> = ({ routes, onClose }) => {
...
+
+  const {
+    trigger,
+    formState: { isValid },
+  } = useFormContext();
+
+  const handleNext = (event: MouseEvent<HTMLButtonElement>) => {
+    event?.preventDefault();
+    trigger();
+    isValid && handleStep(step + 1);
```
Comment (yehorGuyAvrahamCom) [inline src/Components/CheckoutModal/ProgressBar/ProgressControls.tsx:58]: You can change it to this  
```
const { trigger } = useFormContext();

  const handleNext = (event: MouseEvent<HTMLButtonElement>) => {
    event?.preventDefault();
    trigger().then((isValid) => {
      if (isValid) {
        handleStep(step + 1);
      }
    })
```

File: src/Components/CheckoutModal/Sections/Delivery/AccordionItemComponent.tsx [hunk 8]
```diff
@@ -63,31 +56,38 @@ export const AccordionItemComponent: FC<IAccordionItemComponent> = ({
...
+        >
+          <Stack spacing={5}>
+            {data.map((item) => (
+              <Radio
+                _focus={{ boxShadow: 'unset' }}
+                colorScheme="blackAlpha"
+                key={item.name}
+                __css={{
+                  '.span': { w: 'full' },
+                }}
```
Comment (yehorGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Delivery/AccordionItemComponent.tsx:73]: `_focus`, `colorScheme` and `__css` move to styles object

File: src/Components/CheckoutModal/Sections/Shipping/BoxDivider.tsx [hunk 9]
```diff
@@ -0,0 +1,3 @@
+import { Box } from '@chakra-ui/react';
+
+export const BoxDivider = () => <Box mx="0.5" />;
```
Comment (yehorGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Shipping/BoxDivider.tsx:3]: Remove it and replace this to style `gap="1"` for containers where you use it now

File: src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx [hunk 10]
```diff
@@ -47,37 +50,62 @@ export const FormInput: FC<IFormInput> = ({
...
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
-    const { value, name } = e.target;
-    setInputValue({
-      ...inputValue,
-      [name]: value,
-    });
-    setDeliveryAddress(inputValue.StreetAddress);
-  };
+  const border = () => {
+    if (errors[name]?.message) {
```
Comment (yehorGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx:54]: Create a variable `const errorMessage = errors[name]?.message` and use it

File: src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx [hunk 11]
```diff
@@ -47,37 +50,62 @@ export const FormInput: FC<IFormInput> = ({
...
 
   const colorFocus = isFocus ? 'blackAlpha.900' : 'blackAlpha.600';
 
   const backgroundColorFocus = isFocus ? 'white' : 'checkoutInputBg';
 
   const placeholderFocus = isFocus ? undefined : label;
 
-  const pt = inputRef.current?.value || isFocus ? '3' : '1';
+  const pt =
+    errors[name]?.message || inputRef.current?.value || isFocus ? '3' : '1';
```
Comment (yehorGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx:72]: Move it all into separated styles function

File: src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx [hunk 12]
```diff
@@ -47,37 +50,62 @@ export const FormInput: FC<IFormInput> = ({
...
       display="flex"
       backgroundColor={backgroundColorFocus}
       borderRadius="md"
       px="4"
       pt={pt}
-      border={borderFocus}
+      border={border()}
       w={w}
       mb="1"
       h="12"
```
Comment (yehorGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx:106]: All of this can be moved to styles function what I wrote above

File: src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+  };
+
+  return (
+    <>
+      <Flex flexWrap="wrap">
+        <Flex direction={isMobile ? 'row' : 'column'} w="full">
+          <FormInput
+            pattern={{
+              value:
+                /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
```
Comment (yehorGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx:44]: Make a variable for it

File: src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+          name="marketing"
+        >
+          <Text fontWeight="400">
+            {massages.marketing}
+            <Text as="label" fontWeight="bold">
+              Terms
+            </Text>
+            <Text as="label"> and </Text>
+            <Text as="label" fontWeight="bold">
+              Privacy Policy
```
Comment (AvnerGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx:166]: move to massages

File: src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+      <Box mt="3">
+        <CustomCheckbox
+          formlabelstyle={FormLabelStyles}
+          variant={EVariantCheckBox.Square}
+          name="marketing"
+        >
+          <Text fontWeight="400">
+            {massages.marketing}
+            <Text as="label" fontWeight="bold">
+              Terms
```
Comment (AvnerGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx:162]: move to massages

**Review by yehorGuyAvrahamCom — COMMENTED** [review]
Your code is staying better!

File: src/data/input.ts [hunk 16]
```diff
@@ -1,46 +1,46 @@
 export const inputData = {
   shippingPart1: [
     {
-      name: 'EmailAddress',
```
Comment (AvnerGuyAvrahamCom) [inline src/data/input.ts:4]: Delete this file

Comment (AvnerGuyAvrahamCom) [reply src/data/input.ts:4]: done

File: src/config.ts [hunk 17]
```diff
@@ -32,3 +32,6 @@ export enum EVariantCheckBox {
   Circle = 'circle',
   Square = 'square',
 }
+
+export const GOOGLE_API = 's';
+// process.env.GOOGLE_API_KEY;
```
[the remark this answers is not in the document]
Comment (AvnerGuyAvrahamCom) [reply src/config.ts:37]: done

File: src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx (hunk 10 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx:54]: done

File: src/Components/CheckoutModal/OrderSummary/index.tsx (hunk 6 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/OrderSummary/index.tsx:57]: done

File: src/Components/CheckoutModal/OrderSummary/index.tsx (hunk 5 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/OrderSummary/index.tsx:49]: done

File: src/Components/CheckoutModal/ProgressBar/ProgressControls.tsx (hunk 7 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/ProgressBar/ProgressControls.tsx:58]: done

File: src/Components/CheckoutModal/Sections/Delivery/AccordionItemComponent.tsx (hunk 8 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Delivery/AccordionItemComponent.tsx:73]: done

File: src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx (hunk 13 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx:44]: done

File: src/Components/CheckoutModal/index.tsx [hunk 18]
```diff
@@ -18,25 +24,36 @@ const CheckoutModal = () => {
...
 
   return (
     <Box h="full">
-      <Button onClick={handleClick}>Open Modal</Button>
+      <Flex direction="column">
+        <Button onClick={() => handleClick('ORIGINAL')}>Open Modal</Button>
+        <Button onClick={() => handleClick('NO_VALIDATION')}>
+          Open Modal with NO Validation
+        </Button>
+        <Button onClick={() => handleClick('PRE_FILL')}>
```
[the remark this answers is not in the document]
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/index.tsx:47]: done

File: src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx (hunk 15 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx:162]: done

File: src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx (hunk 14 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx:166]: done

File: src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx (hunk 12 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx:106]: done

File: src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx (hunk 11 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx:72]: done

File: src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx (hunk 12 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx:106]: done

File: src/Components/CheckoutModal/Sections/Shipping/BoxDivider.tsx (hunk 9 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/BoxDivider.tsx:3]: i don’t always  use this gap, because of that i didnt do it

File: src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx [hunk 19]
```diff
@@ -47,37 +83,35 @@ export const FormInput: FC<IFormInput> = ({
...
 
-  const backgroundColorFocus = isFocus ? 'white' : 'checkoutInputBg';
-
   const placeholderFocus = isFocus ? undefined : label;
 
-  const pt = inputRef.current?.value || isFocus ? '3' : '1';
+  const modalVariant = useRecoilValue(modalVariants);
+
+  const isValidationRequired = (required?: boolean) => {
+    if (modalVariant === 'NO_VALIDATION') {
```
Comment (yehorGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx:95]: Type?) Enum/conts?

File: src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx [hunk 20]
```diff
@@ -0,0 +1,171 @@
...
+      <Flex flexWrap="wrap">
+        <Flex direction={isMobile ? 'row' : 'column'} w="full">
+          <FormInput
+            message={massages.emailAddressRequired}
+            name="emailAddress"
+            label={massages.emailAddress}
+            type="email"
+            w={!isMobile ? 'full' : 'unset'}
+          />
+          {isMobile && <BoxDivider />}
```
Comment (yehorGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx:47]: You didn't remove `<BoxDivider />`

Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx]: done

File: src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx (hunk 19 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/FormInput.tsx]: done

File: src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx [hunk 21]
```diff
@@ -0,0 +1,171 @@
...
+          formlabelstyle={FormLabelStyles}
+          variant={EVariantCheckBox.Square}
+          name="marketing"
+        >
+          <Text fontWeight="400">
+            {massages.marketing}
+            <Text as="label" fontWeight="bold">
+              {massages.terms}
+            </Text>
+            <Text as="label"> and </Text>
```
Comment (yehorGuyAvrahamCom) [inline src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx:155]: Move "and" to messages

**Review by yehorGuyAvrahamCom — COMMENTED** [review]
Last one )

File: src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx (hunk 21 above)
Comment (AvnerGuyAvrahamCom) [reply src/Components/CheckoutModal/Sections/Shipping/ShippingForm.tsx]: done

**Review by yehorGuyAvrahamCom — APPROVED** [review]
